ynr/apps/elections/uk/every_election.py

```diff
@@ -186,6 +186,50 @@ def get_or_create_post_election(self, parent):
             )
         return (self.post_election_object, self.post_election_created)
 
+    def delete_post_election(self):
+        try:
+            pee = PostExtraElection.objects.get(
+                ballot_paper_id=self["election_id"]
+            )
+        except PostExtraElection.DoesNotExist:
+            # if it doesn't already exist, this might be because:
+            # - we deleted it in EE before it was ever imported into YNR
+            # - we already deleted it
+            # either way, we wanted it gone and now its not there
+            return
+
+        pee.delete()
```

Comment:
It might be worth using a collector like the Django admin interface does here. This would catch any related object that might get deleted.

See [this code](https://github.com/DemocracyClub/yournextrepresentative/blob/master/ynr/apps/candidates/models/popolo_extra.py#L47) for an example of that.

See other comment about `save_delete` here too.

Reply:
Right, so whereas I'm just checking `pee.membership_set.all().exists()` to explicitly check for related memberships, rewriting like that will allows us to check for any relationships (including ones we might create in future).. that sounds better :+1: Will have a proper delve into that tomorrow.